### PR TITLE
Issue 611 - scale units to units provided

### LINKED ISSUE #611: Take in a units value from configuration file and apply it to the model import where possible
### Description
There can be a discrepancy between the file's unit models and the model settings units due to user errors (or in the case of IFC, it can be non deterministic depending on the way it is exported)

For BIM file formats we can pull out the units from the file, so we should take into account of what is our model setting units and apply a scaling so it matches

Where it is impossible (e.g. obj), there should be no scaling applied.

### Goals
- [x] Read in a units from the model import config file and apply scaling
- [x] Do not apply scaling if the file does not have units information

### Tasks
- [x] Add code to support reading units in from the configuration file
  - [x] Thread this information through to the model importer
- [x] Add new virtual function for getting units from the abstract importer
  - [x] Implement the response for all applicable importer types
    - [x] .bim - unknown (We don't record units)
    - [x] .spm - can be found in worldSettings
    - [x] .ifc - will always be in metres as per #600 
    - [x] assimp - Unknown (Most file formats we support through this is unit less anyway [fbx is also unitless](https://forums.autodesk.com/t5/fbx-forum/fbx-units-and-scaling/m-p/9811613/highlight/true#M10079))
    - [x] ODA 
      - [x] rvt
      - [x] dgn/dwg
      - [x] navis
- [x] Add scaling logic onto the importer
- [x] Create a pairing issue for 3drepo.io to make necessary changes

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #611
#### Description
- This issue relies on 3drepo.io passing down the units (requires https://github.com/3drepo/3drepo.io/pull/4080)
- This units is parsed and written in ModelImportConfig
- the model will be scaled to the units if we know the file units, otherwise, this step is not performed

#### Test cases
model imports should now be scaled to the units provided

### CODE REVIEW

**Review by sebjf — COMMENTED** [review]
I made three trivial comments, but looks & works well as far as I can tell. The APIs are in a good place for the mesh optimisation.

File: bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/file_processor_nwd.cpp [hunk 1]
```diff
@@ -69,6 +72,24 @@ uint8_t FileProcessorNwd::readFile()
 			throw new OdError("Navisworks Composite/Index files (.nwf) are not supported. Files must contain embedded geometry (.nwd)");
 		}
 
+		switch (pNwDb->getUnits()) {
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/file_processor_nwd.cpp:75]: yes fair point!

File: bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_abstract.h [hunk 2]
```diff
@@ -69,6 +68,14 @@ namespace repo {
 				virtual bool applyReduction() const { return true; }
 				virtual bool requireReorientation() const { return false; }
 
+				/**
+				* Retrieve the filename from a full path
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_abstract.h:72]: ... yes 😆

**Review by sebjf — APPROVED** [review]

**Comment by carmenfan** [comment]
Product approved https://github.com/3drepo/3drepo.io/pull/4080#pullrequestreview-1355073024

**Comment by carmenfan** [comment]
Approved as per https://github.com/3drepo/3drepo.io/pull/4080

**Pull request merged**